src/mono/mono/mini/debugger-agent.c

```diff
@@ -7624,6 +7655,70 @@ collect_interfaces (MonoClass *klass, GHashTable *ifaces, MonoError *error)
 	}
 }
 
+static int get_static_field_value(MonoClassField* f, MonoClass* klass, MonoDomain* domain, MonoInternalThread* thread, Buffer* buf)
+{
+	MonoStringHandle string_handle = MONO_HANDLE_NEW(MonoString, NULL); // FIXME? Not always needed.
+	ERROR_DECL(error);
+	guint8* val;
+	MonoVTable* vtable;
+	MonoClass* k;
+	guint32 special_static_type;
+	gboolean found;
+
+	if (!(f->type->attrs & FIELD_ATTRIBUTE_STATIC))
+		return -1;
+
+	special_static_type = mono_class_field_get_special_static_type(f);
+	if (special_static_type != SPECIAL_STATIC_NONE) {
+		if (!(thread && special_static_type == SPECIAL_STATIC_THREAD))
+			return -1;
+	}
+
+	/* Check that the field belongs to the object */
+	found = FALSE;
+	for (k = klass; k; k = m_class_get_parent(k)) {
+		if (k == f->parent) {
+			found = TRUE;
+			break;
+		}
+	}
+	if (!found)
+		return -1;
+
+	vtable = mono_class_vtable_checked(f->parent, error);
+	if (!is_ok(error))
+		return -1;
+
+	if (agent_config.using_icordbg)
```

Comment:
can this condition be changed to use a function argument instead?  Overall I'd like to minimize the number of dependencies on global flags like this.

Reply:
Changed as we discussed in the other PR, to be a parameter of protocol version.